### PR TITLE
feat(components/molecule/tabs): make tabs accesibles

### PR DESCRIPTION
**Pull request by oriolpuig**
### Types of changes

- [ ] 🪲 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🧾 Documentation
- [ ] 📷 Demo
- [x] 🧪 Test
- [ ] 🧠 Refactor
- [ ] 💄 Styles

### Description, Motivation and Context
Currently our component is not Accessibility friendly, with this PR we want to make it possible. 
We use the specific role for each element to improve User Experience and our Test Experience with Testing Library.

#### 📃 DOCS: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role

### CODE REVIEW

**Review by andresz1 — APPROVED** [review]

**Comment by andresz1** [comment]
Nice PR 👏🏻👏🏻👏🏻 Should we also add [aria-orientation](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-orientation) when the `type` is set to `vertical` ? https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tablist_role

File: components/molecule/tabs/src/components/MoleculeTabs.js [hunk 1]
```diff
@@ -42,20 +42,28 @@ const MoleculeTabs = ({
 
   const activeTabContent = childrenArray.reduce((activeContent, child) => {
     if (child) {
-      const {children: childrenChild, active} = child.props
-      return active ? childrenChild : activeContent
+      const {children: childrenChild, active, numTab} = child.props
+      return active ? (
+        <div
+          className={CLASS_CONTENT}
+          id={`molecule-tab-content-${numTab}`}
```
Comment (andresz1) [inline components/molecule/tabs/src/components/MoleculeTabs.js]: If we add two tabs in the same page. Is this id going to be repeated ?

Comment (oriolpuig) [reply components/molecule/tabs/src/components/MoleculeTabs.js]: For sure, yes. Good catch. 
I've added this to help the `aria-controls` on tab elements 🤔. 

In order to do it unique, we must pass a `id` to the MoleculeTab and use it to build it dynamically, isn't it? What do you think?

Comment (andresin87) [reply components/molecule/tabs/src/components/MoleculeTabs.js]: Pass an id to `MoleculeTabs`, prop drill it to the MoleculeTab and concat the numtab to get the id of each.

File: components/molecule/tabs/src/components/MoleculeTabs.js [hunk 2]
```diff
@@ -36,26 +38,39 @@ const MoleculeTabs = ({
...
+        aria-orientation={
+          isVerticalOrientation ? TYPES.VERTICAL : TYPES.HORIZONTAL
+        }
+      >
         {extendedChildren}
       </ul>
-      {activeTabContent ? (
-        <div className={CLASS_CONTENT}>{activeTabContent}</div>
-      ) : null}
+      {activeTabContent || null}
```
Comment (andresin87) [inline components/molecule/tabs/src/components/MoleculeTabs.js]:
```suggestion
      {activeTabContent}
```

File: components/molecule/tabs/src/components/MoleculeTabs.js [hunk 3]
```diff
@@ -36,26 +38,39 @@ const MoleculeTabs = ({
...
-      return active ? childrenChild : activeContent
+      const {children: childrenChild, active, numTab} = child.props
+
+      return active ? (
+        <div className={CLASS_CONTENT} id={`${id}-${numTab}`} role="tabpanel">
+          {childrenChild}
+        </div>
+      ) : (
+        activeContent
+      )
```
Comment (andresin87) [inline components/molecule/tabs/src/components/MoleculeTabs.js]:
```js
return (<div className={CLASS_CONTENT} id={`${id}-${numTab}`} role="tabpanel">{active ? childrenChild : activeContent}</div>)
```
retrocompatibility

Comment (andresin87) [reply components/molecule/tabs/src/components/MoleculeTabs.js]: review plz, maybe its wrong

Comment (oriolpuig) [reply components/molecule/tabs/src/components/MoleculeTabs.js]: Nope, it's not well. Your solution add the active tab content inside another active tab content

image

Comment (andresin87) [reply components/molecule/tabs/src/components/MoleculeTabs.js]:
```js
if(active) {
  return (
    <div className={CLASS_CONTENT} id={`${id}-${numTab}`} role="tabpanel">
      {childrenChild}
    </div>
  )
}
```

**Review by andresin87 — APPROVED** [review]

**Pull request merged**